pkg/controller/machinepool/ibmcloudactuator_test.go

```diff
@@ -0,0 +1,111 @@
+package machinepool
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	corev1 "k8s.io/api/core/v1"
+
+	machineapi "github.com/openshift/api/machine/v1beta1"
+	ibmcloudprovider "github.com/openshift/cluster-api-provider-ibmcloud/pkg/apis/ibmcloudprovider/v1beta1"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	hivev1ibmcloud "github.com/openshift/hive/apis/hive/v1/ibmcloud"
+	mockibm "github.com/openshift/hive/pkg/ibmclient/mock"
+)
+
+func TestIBMCloudActuator(t *testing.T) {
+	tests := []struct {
+		name                       string
+		clusterDeployment          *hivev1.ClusterDeployment
+		pool                       *hivev1.MachinePool
+		expectedMachineSetReplicas map[string]int64
+		expectedErr                bool
+	}{
+		{
+			name:              "generate machineset",
```

Comment:
So this is only testing one green path. Do we care to add tests for some of the error paths, or other variants (e.g. specified DedicatedHosts, BootVolume, Zones)?

Reply:
Yes, I'll add more tests for those suggestions.

Reply:
Added some more tests for these.